server/lib/models/matchmaking-times.ts

```diff
@@ -0,0 +1,200 @@
+import db from '../db'
+import sql from 'sql-template-strings'
+import { MatchmakingType } from '../../../common/matchmaking'
+
+export interface MatchmakingTime {
+  id: string
+  type: MatchmakingType
+  startDate: Date
+  enabled: boolean
+}
+
+function convertFromDb(props: {
+  /* eslint-disable camelcase */
+  id: string
+  matchmaking_type: MatchmakingType
+  start_date: Date
+  enabled: boolean
+  /* eslint-enable camelcase */
+}): MatchmakingTime {
+  return {
+    id: props.id,
+    type: props.matchmaking_type,
+    startDate: props.start_date,
+    enabled: props.enabled,
+  }
+}
+
+export async function getCurrentMatchmakingTime(
+  matchmakingType: MatchmakingType,
+): Promise<MatchmakingTime | null> {
+  const query = sql`
+    SELECT *
+    FROM matchmaking_times
+    WHERE matchmaking_type = ${matchmakingType} AND start_date <= ${new Date()}
+    ORDER BY start_date DESC
+    LIMIT 1;
+  `
+
+  const { client, done } = await db()
+  try {
+    const result = await client.query(query)
+    return result.rows.length > 0 ? convertFromDb(result.rows[0]) : null
+  } finally {
+    done()
+  }
+}
+
+async function getTotalTimes(
+  matchmakingType: MatchmakingType,
+  date = new Date(),
+  isFuture = true,
+): Promise<number> {
+  const query = sql`
+    SELECT COUNT(id)
+    FROM matchmaking_times
+    WHERE matchmaking_type = ${matchmakingType} AND start_date
+  `
+    .append(isFuture ? '>' : '<')
+    .append(sql`${date};`)
+
+  const { client, done } = await db()
+  try {
+    const result = await client.query(query)
+    return parseInt(result.rows[0].count, 10)
```

Comment:
I don't think this is a string?

(You seem to have done this and the `COUNT(id)` thing in a few places in other models, we should fix those as well)

Reply:
Hmm, `typeof` does detect it as a `string` from what I can tell.

Reply:
Yeah I guess that is actually the case:

> Unfortunately those are going to return strings now since the return type of the COUNT operator is bigint which, by definition, can exceed the maximum value of an int in JavaScript.

We probably need to fix the other places that use this now, where it seems to work by accident. I guess it looks like just email verifications now that I look at it. `invites` returns a badly typed thing from one of its queries but fixes it at some other random point, but that's soon-to-be-deleted code anyway.